### PR TITLE
Azure SQL Management - Client Credentials 

### PR DESCRIPTION
**Pull request by noydavidi**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes:[link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-10641)

## Description
Added Client Credentials Flow for authentication

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by ShirleyDenkberg** [comment]
@michal-dagan @JudahSchwartz Doc review completed.  Looks OK.

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md [hunk 1]
```diff
@@ -38,6 +38,29 @@ Follow these steps for a self-deployed configuration:
...
+To configure a Microsoft integration that uses this authorization flow with a self-deployed Azure application:
+   1. In the **Authentication Type** field, select the **Client Credentials** option.
+   2. In the **Application ID** field, enter your Client/Application ID.
+   3. In the **Tenant ID** field, enter your Tenant ID .
+   4. In the **Client Secret** field, enter your Client Secret.
+   5. Click **Test** to validate the URLs, token, and connection
+   6. Save the instance.
+
+### Testing authentication and connectivity
+If you are using Device Code Flow or Authorization Code Flow, for testing your authentication and connectivity to the Azure DevOps service run the ***!azure-devops-auth-test*** command.
```
Comment (michal-dagan) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/README.md:61]: fix Azure DevOps references

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/AzureSQLManagement_description.md [hunk 2]
```diff
@@ -52,3 +60,21 @@ Follow one of these steps for authentication based on Azure Managed Identities:
...
+   1. In the **Authentication Type** field, select the **Client Credentials** option.
+   2. In the **Application ID** field, enter your Client/Application ID.
+   3. In the **Tenant ID** field, enter your Tenant ID .
+   4. In the **Client Secret** field, enter your Client Secret.
+   5. Click **Test** to validate the URLs, token, and connection
+   6. Save the instance.
+
+### Testing authentication and connectivity
+
+If you are using Device Code Flow or Authorization Code Flow, for testing your authentication and connectivity to the Azure DevOps service run the ***!azure-devops-auth-test*** command.
```
Comment (michal-dagan) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/AzureSQLManagement_description.md:79]: fix Azure DevOps references

**Review by JudahSchwartz — APPROVED** [review]

File: Packs/AzureSQLManagement/Integrations/AzureSQLManagement/AzureSQLManagement_description.md [hunk 3]
```diff
@@ -52,3 +60,21 @@ Follow one of these steps for authentication based on Azure Managed Identities:
...
+   1. In the **Authentication Type** field, select the **Client Credentials** option.
+   2. In the **Application ID** field, enter your Client/Application ID.
+   3. In the **Tenant ID** field, enter your Tenant ID .
+   4. In the **Client Secret** field, enter your Client Secret.
+   5. Click **Test** to validate the URLs, token, and connection
+   6. Save the instance.
+
+### Testing authentication and connectivity
+
+If you are using Device Code Flow or Authorization Code Flow, for testing your authentication and connectivity to the Azure SQL Management service run the ***!azure-devops-auth-test*** command.
```
Comment (michal-dagan) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/AzureSQLManagement_description.md:79]:
```suggestion
If you are using Device Code Flow or Authorization Code Flow, for testing your authentication and connectivity to the Azure SQL Management service run the ***!azure-sql-auth-test*** command. 
```
```suggestion
If you are using Device Code Flow or Authorization Code Flow, for testing your authentication and connectivity to the Azure SQL Management service run the ***!azure-devops-auth-test*** command. 
```

Comment (michal-dagan) [inline Packs/AzureSQLManagement/Integrations/AzureSQLManagement/AzureSQLManagement_description.md:79]:
```suggestion
If you are using Device Code Flow or Authorization Code Flow, for testing your authentication and connectivity to the Azure SQL Management service run the ***!azure-sql-auth-test*** command. 
```

**Pull request merged**